### DISCUSSION TITLE
Auto generated docs menu

### DISCUSSION THREAD

**Discussion by muslimpribadi** [thuliteio/doks#402] [Q&A] [answered]
Hi, I'm planning to have a 100+ pages of documentation that will be under docs/ directory,
can the `docs menu` auto generated from docs/ content tree directory?
..
├── blog/
├── contributors/
├── docs/
└── _index.md

Great theme, it is fast indeed!

**Comment by h-enk** [comment, marked as answer]
@muslimpribadi, a little late, but the documentation menu needs to be set up like so — see [Add to docs menu](https://getdoks.org/tutorial/add-menus/#add-to-docs-menu). 

So, if you want to automate things, you'll need to write a script adding second level menu items in the frontmatter of a documentation page, e.g.:

```md
..
menu:
  docs:
    parent: "prologue"
weight: 010
..
```

**Reply by kuncarahadi** [reply]
@h-enk how about using taxonomies list? I tried using front-matter tags/taxo to organize the menu.

```HTML
{{$current := .}}
<section >
    <ul class="list-unstyled">
        {{ range $key, $taxonomy := .Site.Taxonomies.tags }}
        <li><h3>{{ $key }}</h3></li>
        <ul class="list-unstyled">
            {{ range $taxonomy.Pages }}
               <li>
                <a class="docs-link{{ if eq .File.UniqueID $current.File.UniqueID }} active{{ end }}" href="{{ .Permalink }}">{{ .Title }}</a>
              </li>
            {{ end }}
        </ul>
        {{ end }}
    </ul>
</section>
```

I'm not sure about the aesthetic, but i think it's more effective for much docs.  looks like this https://doks-taxo.netlify.app/docs-taxo/how-to-update-2/

**Reply by h-enk** [reply]
Clever /good thinking /I like this! I will be looking into this — today

**Reply by h-enk** [reply]
PR https://github.com/h-enk/doks/pull/617 implements the option to auto generate the menu from the directory-folder (tree) structure (no manual configuration needed, respects set weight, respects setting `mainSections`) 

I'll keep the tag based setup in mind — possibly for a later date
